### PR TITLE
Added feature for excluding device from a queue

### PR DESCRIPTION
**Pull request by danielfinke**
Move maybe_exclude_from_queues function calls to vars

#1248 against master

### CODE REVIEW

**Comment by lazedo** [comment]
@danielfinke can you provide a use case for this ?
i see an apent logging in to a device and not getting any calls and not being aware of that ?

**Comment by danielfinke** [comment]
@lazedo we have a few clients who have a desk phone and a cell phone. They also are not dedicated agent staff; they sometimes have other roles. They can log out of acdc when they leave their desk, but when they are at their desk they want to only receive calls to their desk phone.

File: applications/callflow/src/cf_endpoint.erl [hunk 1]
```diff
@@ -661,6 +662,18 @@ maybe_do_not_disturb(Endpoint, _, _) ->
             {'error', 'do_not_disturb'}
     end.
 
+-spec maybe_exclude_from_queues(wh_json:object(), wh_json:object(), whapps_call:call()) ->
+                                        'ok' |
+                                        {'error', 'exclude_from_queues'}.
+maybe_exclude_from_queues(Endpoint, _, Call) ->
+    QueueId = whapps_call:custom_channel_var(<<"Queue-ID">>, Call),
```
Comment (jamesaimonetti) [inline applications/callflow/src/cf_endpoint.erl]: Can we do something to make this short-circuit faster?

```erlang
case is_binary(whapps_call:custom_channel_var(<<"Queue-ID">>, Call))
    andalso wh_json:is_true(<<"exclude_from_queues">>, Endpoint)
of
    'false' -> 'ok';
    'true' -> {'error', 'excluded'}
end.
```

Something like this, to me, reads better as to the intention of the code (versus mashing the two toggles into a tuple).

Thoughts?

**Comment by danielfinke** [comment]
@jamesaimonetti I am certainly happy with that suggestion - I've pushed a new commit

**Comment by jamesaimonetti** [comment]
@danielfinke Thanks!

**Pull request merged**